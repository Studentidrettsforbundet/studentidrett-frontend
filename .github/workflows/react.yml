on:
  pull_request:
    branches:
      - dev

name: Frontend build and push to Azure

jobs:
    build-and-deploy:
        env:
          DJANGO_SECRET_KEY: ${{ secrets.DJANGO_SECRET_KEY }}
          IMAGE_VERSION: ${{ github.actor }}
          ACR: ${{ secrets.REGISTRY_LOGIN_SERVER }}

        runs-on: ubuntu-latest
        steps:
        - name: 'Checkout GitHub Action'
          uses: actions/checkout@master

        - name: 'Login via Azure CLI'
          uses: azure/login@v1
          with:
            creds: ${{ secrets.AZURE_CREDENTIALS }}

        - name: 'Build image'
          run: |
            docker build . -t ${{ secrets.REGISTRY_LOGIN_SERVER }}/studentidrett-frontend:dev
            
        - name: 'Push image to ACR'
          uses: azure/docker-login@v1
          with:
            login-server: ${{ secrets.REGISTRY_LOGIN_SERVER }}
            username: ${{ secrets.REGISTRY_USERNAME }}
            password: ${{ secrets.REGISTRY_PASSWORD }}
        - run: |
            docker push ${{ secrets.REGISTRY_LOGIN_SERVER }}/studentidrett-frontend:dev
        - name: Azure logout
          run: |
            az logout
