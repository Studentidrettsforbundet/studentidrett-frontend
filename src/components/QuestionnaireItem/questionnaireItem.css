input[type="radio"] {
}

input[type="radio"] + label span {
  display:inline-block;
  width:19px;
  height:19px;
  margin:-2px 10px 0 0;
  vertical-align:middle;
  background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/210284/check_radio_sheet.png) -38px top no-repeat;
  cursor:pointer;
}

input[type="radio"]:checked + label span {
  background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/210284/check_radio_sheet.png) -57px top no-repeat;
}