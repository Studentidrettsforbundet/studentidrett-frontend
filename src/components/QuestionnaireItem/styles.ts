import { style } from 'typestyle';

export const questionnaireItem = style({
    display: 'flex',
    flexDirection: 'column',
    color: 'rgba(38, 54, 111, 1)',
    background: 'white',
    borderRadius: 4,
    padding: '12px',
    alignItems: 'center',
    textAlign: 'center',
    $nest: {
        span: {
            fontWeight: 700,
        },
    },
    marginBottom: '4px',
    maxWidth: '700px',
});

export const radioButton = style({
    marginTop: '5px',
    $nest: {
        input: {
            display: 'none',
        },
        'input + label': {
            margin: '0',
            height: '0',
        },
        'input + label span': {
            display: 'inline-block',
            border: '3px solid rgba(255, 171, 64, 1)',
            borderRadius: 100,
            cursor: 'pointer',
        },
        '.controller2': {
            height: '24px',
            width: '24px',
        },
        '.controller1, .controller3': {
            height: '26px',
            width: '26px',
        },
        '.controller0, .controller4': {
            height: '32px',
            width: '32px',
        },
        'input:checked + label span': {
            background: 'radial-gradient(rgba(255, 171, 64, 1) 50%, rgba(255, 0, 0, 0) 51%)',
        },
    },
});

export const questionText = style({
    fontSize: '16px',
    fontWeight: 700,
    color: 'black',
    textAlign: 'center',
    marginBottom: '12px',
});

export const labelContainer = style({
    display: 'flex',
    minWidth: '280px',
    justifyContent: 'space-between',
    fontSize: '14px',
    color: '#777777',
    marginBottom: '6px',
});

export const buttonContainer = style({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    minWidth: '280px',
});
