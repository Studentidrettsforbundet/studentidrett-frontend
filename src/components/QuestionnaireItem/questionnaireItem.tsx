import React from 'react';
import { questionnaireItem, radioButton, questionText, radioContainer, buttonContainer } from './styles';
//import './questionnaireItem.css';

export type QuestionnaireItemProps = {
    id: string;
    text: string;
    left: string;
    right: string;
};

const QuestionnaireItem = (props: { item: QuestionnaireItemProps } & { onChange: any; error: string }) => {
    const handleChange = (event: any) => {
        props.onChange(event.target.value);
    };

    const radioButtons = Array.from(Array(5).keys()).map((number) => {
        return (
            <span key={`${props.item.id}-${number}`} className={radioButton}>
                <input
                    className={radioButton}
                    id={`${props.item.id}-${number}`}
                    type="radio"
                    onChange={handleChange}
                    name={`${props.item.id}`}
                    value={number}
                    data-testid={`${props.item.id}-${number}`}
                />
                <label htmlFor={`${props.item.id}-${number}`}>
                    <span className={`controller${number}`} />
                </label>
            </span>
        );
    });

    return (
        <div className={questionnaireItem} key="PLACEHOLDER">
            <span className={questionText}>{props.item.text}</span>
            <div className={radioContainer}>
                <span>{props.item.left}</span>
                <span>{props.item.right}</span>
            </div>
            <div className={buttonContainer}>{radioButtons}</div>
            {props.error && <span style={{ color: 'red' }}>{props.error}</span>}
        </div>
    );
};

export default QuestionnaireItem;
