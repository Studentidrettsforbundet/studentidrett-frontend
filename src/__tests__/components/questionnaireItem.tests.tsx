import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import QuestionnaireItem from '../../components/questionnaireItem';

const mockFunction = jest.fn();
const mockProps = 
  {
      item: {
          id: 'test_id',
          question: 'Test question',
          leftItem: 'Left item',
          rightItem: 'Right item',
      },
      onChange: mockFunction,
      error: '',
  }
;

describe('questionnaireItem', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  describe('snapshot tests', () => {
  
    it('renders correctly with no error', () => {
      const tree = renderer
        .create(<QuestionnaireItem item={mockProps.item} onChange={mockProps.onChange} error={mockProps.error}/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders correctly with error', () => {
      const tree = renderer
        .create(<QuestionnaireItem item={mockProps.item} onChange={mockProps.onChange} error="Test error"/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  })

  it('should invoke callback on event', () => {
    const {getByTestId} = render(<QuestionnaireItem item={mockProps.item} onChange={mockProps.onChange} error={mockProps.error}/>)

    fireEvent.click(getByTestId('test_id-3'))

    expect(mockFunction).toBeCalledWith('3')
  });
})