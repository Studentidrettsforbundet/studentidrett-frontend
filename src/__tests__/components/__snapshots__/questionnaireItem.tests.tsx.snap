// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questionnaireItem snapshot tests renders correctly with error 1`] = `
<div
  className="item"
  style={
    Object {
      "alignContent": "center",
      "padding": "12px",
    }
  }
>
  <label>
    <h4>
      Test question
    </h4>
    <div>
      Left item
      <span
        style={
          Object {
            "padding": "0px 12px",
          }
        }
      >
        <input
          data-testid="test_id-0"
          name="test_id-0"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={0}
        />
        <input
          data-testid="test_id-1"
          name="test_id-1"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={1}
        />
        <input
          data-testid="test_id-2"
          name="test_id-2"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={2}
        />
        <input
          data-testid="test_id-3"
          name="test_id-3"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={3}
        />
        <input
          data-testid="test_id-4"
          name="test_id-4"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={4}
        />
      </span>
      Right item
    </div>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      Test error
    </span>
  </label>
</div>
`;

exports[`questionnaireItem snapshot tests renders correctly with no error 1`] = `
<div
  className="item"
  style={
    Object {
      "alignContent": "center",
      "padding": "12px",
    }
  }
>
  <label>
    <h4>
      Test question
    </h4>
    <div>
      Left item
      <span
        style={
          Object {
            "padding": "0px 12px",
          }
        }
      >
        <input
          data-testid="test_id-0"
          name="test_id-0"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={0}
        />
        <input
          data-testid="test_id-1"
          name="test_id-1"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={1}
        />
        <input
          data-testid="test_id-2"
          name="test_id-2"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={2}
        />
        <input
          data-testid="test_id-3"
          name="test_id-3"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={3}
        />
        <input
          data-testid="test_id-4"
          name="test_id-4"
          onChange={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="radio"
          value={4}
        />
      </span>
      Right item
    </div>
    
  </label>
</div>
`;
